tasks/build.py

```diff
@@ -73,16 +74,17 @@ def build_easystack_from_pr(pr, event_info):
         run += 1
     run_dir = os.path.join(event_dir, 'run_%03d' % run)
     mkdir(run_dir)
+    return ym, pr_id, run_dir
 
-    gh = github.get_instance()
+
+def download_pull_request(pr, arch_target_map, run_dir, cvmfs_customizations):
     # adopting approach outlined in https://github.com/EESSI/eessi-bot-software-layer/issues/17
     # need to use `base` instead of `head` ... don't need to know the branch name
     # TODO rename to base_repo_name?
     repo_name = pr.base.repo.full_name
     log("build_easystack_from_pr: pr.base.repo.full_name '%s'" % pr.base.repo.full_name)
     branch_name = pr.base.ref
     log("build_easystack_from_pr: pr.base.repo.ref '%s'" % pr.base.ref)
-
     jobs = []
     for arch_target,slurm_opt in arch_target_map.items():
```

Comment:
suggestion: create functions for what's done *in* the loop body, something like:

```python
for arch_target,slurm_opt in arch_target_map.items():
        arch_job_dir = os.path.join(run_dir, arch_target.replace('/', '_'))
        mkdir(arch_job_dir)
        log("arch_job_dir '%s'" % arch_job_dir)

        set_up_pr_in(repo_name, pr_number, arch_job_dir)
        apply_customisations(arch_job_dir, cvmfs_customizations)